### PR TITLE
fix: close GRPC channel when we dispose of clients

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
See https://github.com/googleapis/nodejs-firestore/issues/768, especially https://github.com/googleapis/nodejs-firestore/issues/768#issuecomment-539654138

This is a "Request for comments" PR if this is a viable strategy to dispose of unused GRPC clients, in the hope that this will mitigate memory leaks. If feasible, some of the code in this PR needs to be pushed upstream to the Gapic generator.

### CODE REVIEW

**Comment by schmidt-sebastian** [comment]
FYI @thebrianchen

File: dev/src/index.ts [hunk 1]
```diff
@@ -401,7 +401,8 @@ export class Firestore {
 
         logger('Firestore', null, 'Initialized Firestore GAPIC Client');
         return client;
-      }
+      },
+      /* clientDestructor= */ (client: GapicClient) => client.close()
```
Comment (alexander-fenster) [inline dev/src/index.ts]: hmmm - do we really have `.close()`? :)

Comment (schmidt-sebastian) [reply dev/src/index.ts]: Not yet - but this PR shows how we could add it. We still have to add it to the Gapic generator though (if you are okay with the general direction).

Comment (schmidt-sebastian) [reply dev/src/index.ts]: @alexander-fenster Do you know what the next sets of steps would be if we wanted to add this support?

**Comment by schmidt-sebastian** [comment]
@alexander-fenster PR rebased against the Typescript branch and updated.

**Pull request merged**